docs/alerting-recording.md

```diff
@@ -0,0 +1,65 @@
+# Alerting & Recording Rules
+
+### Alerting Rules
+
+Alerting rules are used to trigger alerts based on the violation of any condition(s). These alerts are fired to external services like slack, mails, etc by the [alert manager](https://prometheus.io/docs/alerting/latest/alertmanager/). Alerting rules are written in a YAML file and paths of these files are mentioned in the Prometheus configuration respectively. It is important to note that the evaluation of these conditional rules are performed at the Prometheus side.
+
+```
+groups:
+  - name: <alert-group-name>
+    rules:
+    - alert: <alert-name>
+      expr: <promql_expression>
+      for: <time-interval for how long this to happen to happen to fire an alert>
+      labels:
+        <key>:<value>
+      annotations:
+        summary: <text>
+        description: <description on alert>
+```
+
+Load the above defined alert rule file to your prometheus configuration for evaluation.
+
+```
+rule_files:
+  - "<alert-rules-file>"
+```
+
+Configure the alert manager in prometheus configuration.
+
+```
+alerting:
+  alertmanagers:
+  - static_configs:
+    - targets:
+      - localhost:9093
+```
+
+More details on alerting rules can be found [here](https://prometheus.io/docs/prometheus/latest/configuration/alerting_rules/).
+
+
+### Recording Rules
+
+The recording rules are for pre-calculating frequently used or computationally expensive queries. The results of those rules are saved into their own time series. The newly formed time-series will follow the name of the record. In promscale the rules are evaluated in prometheus i.e define recording rules in prometheus configuration.
+
+Adding recording rules:
+
+```
+groups:
+  - name: my-rules
+    rules:
+    - record: <RECORD_NAME>
+      expr: <PROMQL_EXPR>
+      
+```
+
+Load the above defined recording rule file to your prometheus configuration for evaluation.
+
+```
+rule_files:
+  - "<recording-rules-file>"
+```
+
+More details on recording rules can be found [here](https://prometheus.io/docs/prometheus/latest/configuration/recording_rules/).
+
+**Note**: If `read_recent: true` is set in prometheus configuration the data is fetched from promscale for evaluation. If `read_recent` is disabled the data stored in prometheus will be used for evaluation this needs prometheus retention to be configured higher than the alerting & recording rules evaluation interval for the rules evaluation to happen before dropping the samples. The new alerting & recording rules series are written to promscale.  
```

Comment:
Let's mention here that `read_recent: true` is the recommended way.

Reply:
This statement can be modified to sound better. Also, the second sentence is too long.